monitoring.tf

```diff
@@ -0,0 +1,42 @@
+resource "aws_elasticache_subnet_group" "elasticache-shared-subnet-group" {
+  name                 = "elasticache-shared-subnet-group"
+  subnet_ids           = ["${aws_subnet.apps-shared-1a.id}", "${aws_subnet.apps-shared-1c.id}", "${aws_subnet.apps-shared-1d.id}"]
+}
+
+resource "aws_security_group" "sensu-redis-sg" {
+  name                 = "sensu-redis-shared-sg"
+  description          = "Sensu elasticache SG"
+  vpc_id               = "${aws_vpc.apps-shared-vpc.id}"
+}
+
+resource "aws_security_group_rule" "sensu-redis-sg-allowredisfromall" {
+  type                 = "ingress"
+  from_port            = 6379
+  to_port              = 6379
+  protocol             = "tcp"
+  cidr_blocks          = ["${aws_vpc.apps-production-vpc.cidr_block}", "${aws_vpc.apps-staging-vpc.cidr_block}, ${aws_vpc.apps-shared-vpc.cidr_block}"]
+
+  security_group_id    = "${aws_security_group.sensu-redis-sg.id}"
+}
+
+resource "aws_security_group_rule" "sensu-redis-sg-allowegress" {
```

Comment:
No need to allow egress access to the internet. Let's restrict it to the 3 subnets.